src/main/kotlin/org/rust/cargo/project/module/persistence/impl/CargoModuleTargetsServiceImpl.kt

```diff
@@ -0,0 +1,54 @@
+package org.rust.cargo.project.module.persistence.impl
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.openapi.components.StoragePathMacros
+import org.rust.cargo.CargoProjectDescription
+import org.rust.cargo.project.module.persistence.CargoModuleTargetsService
+import java.util.*
+
+@State(
+    name = "CargoModuleTargets",
+    storages = arrayOf(Storage(file = StoragePathMacros.MODULE_FILE))
+)
+class CargoModuleTargetsServiceImpl : PersistentStateComponent<CargoModuleTargetsServiceImpl.State>
+                                    , CargoModuleTargetsService {
+
+    private var myState: State = State()
+
+    // IDEA serializer requires State components to have default constructors,
+    // nullable fields and mutable collections. `null`s in fields signal
+    // serialization failure, so we can't `!!` here safely.
+    data class State(
+        var targets: MutableCollection<Target> = ArrayList()
+    )
+
+    data class Target(
+        var path: String? = null,
+        var kind: CargoProjectDescription.TargetKind? = null
+    ) {
+
+        fun intoTarget(): CargoProjectDescription.Target? {
```

Comment:
>What are the reasons to have both CargoProjectDescription.Target and this Target?

IDEA's XmlSerializer can't deal with NonNull fields yet. So the choice here is either to have a nullable and not nullable versions of Target, or to have a single nullable Target. I am strongly opposed the second one. 

>It's better to move it into `CargoProjectDescription.Target` and made it a factory method instead of converting one.

This `Target` is an implementation detail of the service, so I don't think we should mention it anywhere except in this file. It is public only because get/load state methods are public. Hm, wait, I think I can make it private!